frame/bounties/src/subbounty_migration.rs

```diff
@@ -0,0 +1,163 @@
+pub enum BountyStatus<AccountId, BlockNumber> {
+	/// The bounty is proposed and waiting for approval.
+	Proposed,
+	/// The bounty is approved and waiting to become active at next spend period.
+	Approved,
+	/// The bounty is funded and waiting for curator assignment.
+	Funded,
+	/// A curator has been proposed by the `ApproveOrigin`. Waiting for acceptance from the curator.
+	CuratorProposed {
+		/// The assigned curator of this bounty.
+		curator: AccountId,
+	},
+	/// The bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// An update from the curator is due by this block, else they are considered inactive.
+		update_due: BlockNumber,
+	},
+	/// The bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this bounty.
+		curator: AccountId,
+		/// The beneficiary of the bounty.
+		beneficiary: AccountId,
+		/// When the bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct OldBounty<AccountId, Balance, BlockNumber> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The curator fee. Included in value.
+	fee: Balance,
+	/// The deposit of curator.
+	curator_deposit: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The status of this bounty.
+	status: BountyStatus<AccountId, BlockNumber>,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct Bounty<AccountId, Balance, BlockNumber> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The curator fee. Included in value.
+	fee: Balance,
+	/// The deposit of curator.
+	curator_deposit: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The status of this bounty.
+	status: BountyStatus<AccountId, BlockNumber>,
+	/// active Subbounty count
+	active_subbounty_count: BountyIndex,
+}
+
+struct StorageMigrationBounties;
+impl frame_support::traits::StorageInstance for StorageMigrationBounties {
+	fn pallet_prefix() -> &'static str { "Treasury" }
+	const STORAGE_PREFIX: &'static str = "Bounties";
+}
+#[allow(type_alias_bounds)]
+type Bounties<T: SubBountyMigration> = StorageMap<
+	StorageMigrationBounties,
+	Twox64Concat,
+	BountyIndex,
+	Bounty<T::AccountId, T::Balance, T::BlockNumber>
+>;
+
+/// Apply all of the migrations for SubbountyExtn.
+///
+/// ### Warning
+///
+/// No storage version check, Maybe need check at runtime.
+///
+/// Be aware that this migration is intended to be used only for the mentioned versions. Use
+/// with care and run at your own risk.
+pub fn apply<T: SubBountyMigration>( ) -> Weight {
+
+	let maybe_storage_version = crate_to_pallet_version!();
```

Comment:
FYI you are not really using this, you could bump the crate version here and make sure that the migration only happens on this crate version, similar to `elections-phragmen`.

Reply:
Indeed `maybe_storage_version` is used only for log trace purpose, and migration is forced based on some discussion with @thiolliere .